prime-dotnet-webapi/Services/MetabaseService.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Text;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.IdentityModel.Tokens;
+using System.IO;
+using System.Collections.Generic;
+
+namespace Prime.Services
+{
+    public class MetabaseService : IMetabaseService
+    {
+        public MetabaseService()
+        { }
+
+        public string BuildMetabaseEmbeddedString()
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(PrimeEnvironment.MetabaseApi.Key));
+
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256Signature);
+
+            var header = new JwtHeader(credentials);
+
+            // 10 minutes in the future represented by milliseconds from epoch
+            var expired = DateTime.Now.AddMinutes(10).Subtract(DateTime.MinValue.AddYears(1969)).TotalMilliseconds;
+
+            var payload = new JwtPayload
+           {
+               { "resource", new DashboardResource(PrimeEnvironment.MetabaseApi.DashboardId)},
```

Comment:
Named classes are helpful when de-serializing JSON, but we should be able to use an anonymous class here. Try
` { "resource", new {dashboard = PrimeEnvironment.MetabaseApi.DashboardId} }`